File: python/encryptor.py

class Encryptor:
    def cryptWord(self, word):
        if " " in word:
            raise ValueError()

        new_word = "";
        for i in range(len(word)):
            char_value = ord(word[i])
            new_word += chr( char_value + 2)

        return new_word

    def cryptWordToNumbers(self, word):
        if " " in word:
            raise ValueError()

        new_word = "";
        for i in range(len(word)):
            char_value = ord(word[i])
            new_word += str(char_value + 2)

        return new_word

    def cryptWordWithCharsToReplace(self, word, chars_to_replace):
        if " " in word:
            raise ValueError()
        result = list(word)
        for i in range(len(word)):
            for j in range(len(chars_to_replace)):
                if chars_to_replace[j] == word[i]:
                    char_value = ord(word[i])
                    result[i] = chr( char_value + 2)
        return "".join(result)

    def cryptSentence(self, sentence):
        new_word = "";
        for i in range(len(sentence)):
            char_value = ord(sentence[i])
            new_word += chr( char_value + 2)

        return new_word

    def getWords(self, sentence):
        return sentence.split()

    def printWords(self, sentence):
        words = self.getWords(sentence)
        for word in words:
            print("<%s>" % word)

File: csharp/Encryptor.cs
using System;

public class Encryptor
{
    public static void Main(string[] args)
    {
        var encryptor = new Encryptor();
        encryptor.PrintWords(encryptor.CryptSentence("This is just a test"));
    }

    public String CryptWord(String word)
    {
        if (word.Contains(" "))
            throw new ArgumentException();

        char[] wordArray = word.ToCharArray();
        String newWord = "";
        for (int i = 0; i < word.Length; i++)
        {
            int charValue = wordArray[i];
            newWord += (char) (charValue + 2);
        }

        return newWord;
    }

    public String CryptWordToNumbers(String word)
    {
        if (word.Contains(" "))
            throw new ArgumentException();

        char[] wordArray = word.ToCharArray();
        String newWord = "";
        for (int i = 0; i < word.Length; i++)
        {
            int charValue = wordArray[i];
            newWord += charValue + 2;
        }

        return newWord;
    }

    public String CryptWord(String word, String charsToReplace)
    {
        if (word.Contains(" "))
            throw new ArgumentException();

        char[] wordArray = word.ToCharArray();
        char[] replacement = charsToReplace.ToCharArray();
        char[] result = (char[]) wordArray.Clone();
        for (int i = 0; i < wordArray.Length; i++)
        {
            for (int j = 0; j < replacement.Length; j++)
            {
                if (replacement[j] == wordArray[i])
                {
                    int charValue = wordArray[i];
                    result[i] = (char) (charValue + 2);
                }
            }
        }
        return new String(result);
    }

    public String CryptSentence(String sentence)
    {
        char[] sentenceArray = sentence.ToCharArray();
        String newWord = "";
        for (int i = 0; i < sentence.Length; i++)
        {
            int charValue = sentenceArray[i];
            newWord += (char) (charValue + 2);
        }

        return newWord;
    }

    public String[] GetWords(String sentence)
    {
        return sentence.Split(' ');
    }


    public void PrintWords(String sentence)
    {
        String[] words = GetWords(sentence);
        foreach (String word in words)
        {
            Console.WriteLine("<" + word + ">");
        }
    }
}

